cmip6_downscaling/tests/methods/test_regions.py

```diff
@@ -0,0 +1,77 @@
+import numpy as np
+import xarray as xr
+
+from cmip6_downscaling.methods.regions import combine_outputs, generate_subdomains
+
+
+def example_3d_dataarray_us_domain(val=0):
+    return xr.DataArray(
+        val,
+        dims=['lat', 'lon', 'time'],
+        coords={'lat': np.arange(19, 56, 1), 'lon': np.arange(-133, -61, 2), 'time': np.arange(5)},
+    )
+
+
+def example_3d_data_us_domain():
+    ds = xr.Dataset()
+    for v in ['var1', 'var2']:
+        ds[v] = example_3d_dataarray_us_domain()
+    return ds
+
+
+def test_generate_subdomains():
+    ds = example_3d_data_us_domain()
+
+    region_def = 'ar6'
+    buffer_size = 4
+    subdomains, mask = generate_subdomains(
+        ex_output_grid=ds.isel(time=0), buffer_size=buffer_size, region_def=region_def
+    )
+
+    # make sure the correct regions are there
+    region_codes = list(subdomains.keys())
+    assert sorted(region_codes) == [1, 2, 3, 4, 5, 6, 7, 8]
```

Comment:
since region `28` is the weird one, can we have a special test for that one?

Reply:
good catch! the current definition for region 28 was indeed wrong. added a test and fixed the code.

Reply:
huzzah! 🤸‍♀️